Core/src/org/sleuthkit/autopsy/geolocation/GeoFilterPanel.java

```diff
@@ -0,0 +1,355 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.geolocation;
+
+import java.awt.GridBagConstraints;
+import java.awt.event.ActionListener;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import javax.swing.ImageIcon;
+import javax.swing.SpinnerNumberModel;
+import org.openide.util.NbBundle.Messages;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.DataSource;
+import org.sleuthkit.datamodel.SleuthkitCase;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ *
+ * Panel to display the filter options for geolocation waypoints.
+ */
+class GeoFilterPanel extends javax.swing.JPanel {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = Logger.getLogger(GeoFilterPanel.class.getName());
+
+    private final SpinnerNumberModel numberModel;
+    private final CheckBoxListPanel<DataSource> checkboxPanel;
+
+    /**
+     * Creates new GeoFilterPanel
+     */
+    @Messages({
+        "GeoFilterPanel_DataSource_List_Title=Data Sources"
+    })
+    GeoFilterPanel() {
+        // numberModel is used in initComponents
+        numberModel = new SpinnerNumberModel(10, 1, Integer.MAX_VALUE, 1);
+
+        initComponents();
+
+        // The gui builder cannot handle using CheckBoxListPanel due to its
+        // use of generics so we will initalize it here.
+        checkboxPanel = new CheckBoxListPanel<>();
+        checkboxPanel.setPanelTitle(Bundle.GeoFilterPanel_DataSource_List_Title());
+        checkboxPanel.setPanelTitleIcon(new ImageIcon(getClass().getResource("/org/sleuthkit/autopsy/images/image.png")));
+        checkboxPanel.setSetAllSelected(true);
+
+        GridBagConstraints gridBagConstraints = new GridBagConstraints();
+        gridBagConstraints.gridx = 0;
+        gridBagConstraints.gridy = 3;
+        gridBagConstraints.fill = java.awt.GridBagConstraints.HORIZONTAL;
+        gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+        gridBagConstraints.weightx = 1.0;
+        gridBagConstraints.weighty = 1.0;
+        gridBagConstraints.insets = new java.awt.Insets(0, 15, 0, 15);
+        add(checkboxPanel, gridBagConstraints);
+
+        try {
+            initCheckboxList();
```

Comment:
We need to initalize the list of checkboxes based on the current case at some other time.

If you close the case and open another it will still be populated with the data sources from the original case.  

Also as Richard points out in his email we can't be sure there is an open case when a top component is constructed and this panel is constructed in the top component constructor.